source/geometries/SIPM_QE_setup.cc

```diff
@@ -0,0 +1,126 @@
+// ----------------------------------------------------------------------------
+// nexus | SIPM_QE_setup.cc
+//
+// Geometry of a set-up to extract the quantum efficiency of a SIPM.
+//
+// The NEXT Collaboration
+// ----------------------------------------------------------------------------
+
+#include "SIPM_QE_setup.h"
+#include "MaterialsList.h"
+#include "BoxPointSampler.h"
+#include "OpticalMaterialProperties.h"
+
+#include <G4Box.hh>
+#include <G4Tubs.hh>
+#include <G4LogicalVolume.hh>
+#include <G4PVPlacement.hh>
+#include <G4Material.hh>
+#include <G4OpticalSurface.hh>
+#include <G4LogicalSkinSurface.hh>
+#include <G4VisAttributes.hh>
+#include <G4NistManager.hh>
+#include <Randomize.hh>
+#include <G4GenericMessenger.hh>
+
+#include <CLHEP/Units/SystemOfUnits.h>
+
+namespace nexus {
+
+  using namespace CLHEP;
+
+  SIPM_QE_setup::SIPM_QE_setup(): BaseGeometry()
+  {
+    msg_ = new G4GenericMessenger(this, "/Geometry/SIPM_QE_setup/",
+				"Control commands of geometry SIPM_QE_setup.");
+
+    G4GenericMessenger::Command& dist_cmd =
+      msg_->DeclareProperty("z_dist", z_dist_,
+			    "Distance of the generation point in z from the surface of the SIPM window.");
+    dist_cmd.SetUnitCategory("Length");
+    dist_cmd.SetParameterName("z_dist", false);
+    dist_cmd.SetRange("z_dist>0.");
+  }
+
+
+
+  SIPM_QE_setup::~SIPM_QE_setup()
+  {
+  }
+
+
+
+  void SIPM_QE_setup::Construct()
+  {
+    // CHAMBER ///////////////////////////////////////////////////////
+
+    // The parameterized width, height and length are internal dimensions.
+    // The chamber thickness is added to obtain the external (total) size.
+    G4double width = 5.*cm;
+    G4double height = 5.*cm;
+    length_ = 5.*cm;
+    G4double thickn = 1.*cm;
+    G4double X = width  + 2.*thickn;
+    G4double Y = height + 2.*thickn;
+    G4double Z = length_ + 2.*thickn;
+
+    G4Box* chamber_solid = new G4Box("CHAMBER", X/2., Y/2., Z/2.);
+
+    G4Material* copper = G4NistManager::Instance()->FindOrBuildMaterial("G4_Cu");
+
+    G4LogicalVolume* chamber_logic =
+      new G4LogicalVolume(chamber_solid, copper, "CHAMBER");
+    this->SetLogicalVolume(chamber_logic);
+
+    // GAS ///////////////////////////////////////////////////////////
+    G4Box* gas_solid = new G4Box("GAS", width/2., height/2., length_/2.);
+
+    // // xenon
+    // G4double gxe_pressure = 10*bar;
+    // G4Material* gxe = MaterialsList::GXe(gxe_pressure);
+    // gxe->SetMaterialPropertiesTable(OpticalMaterialProperties::GXe(gxe_pressure));
+    // G4LogicalVolume* gas_logic = new G4LogicalVolume(gas_solid, gxe, "GAS");
+
+    // air
+    G4Material* air = G4NistManager::Instance()->FindOrBuildMaterial("G4_AIR");
+    air->SetMaterialPropertiesTable(OpticalMaterialProperties::Vacuum());
+    G4LogicalVolume* gas_logic = new G4LogicalVolume(gas_solid, air, "GAS");
+    new G4PVPlacement(0, G4ThreeVector(0,0,0), gas_logic, "GAS",
+		      chamber_logic, false, 0, true);
```

Comment:
Please, set the variable that checks volume overlap to `false`. We want to avoid the check and the output every time we run a geometry.